### PR TITLE
Clean up sccache usage

### LINKED ISSUE #425: Disable sccache if GCS key not available
Secrets are not available to forks or dependabot pull requests. If the GCS key is not available don't enable sccache setup. Having sccache enabled for these type of runs increases build time.

### LINKED ISSUE #595: Increase timeouts when dependabot PR
Dependabot PRs don't have access to secrets so they can take a while to run. Increase timeouts when GitHub actor is dependabot

### LINKED ISSUE #597: Enable failover mode for sccache network issues
Add `SCCACHE_IGNORE_SERVER_IO_ERROR=1` to not fail builds on sccache network issues.

### PR DESCRIPTION
**Pull request by devpow112**
Cleans up hot sccache GCS set-up is handled. Doesn't do the set-up for dependabot PR's. Does read only for PR's and read/write for `master`. Resolves #595, resolves #425 and resolves #597.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -136,3 +136,5 @@ jobs:
       changes-externals: ${{needs.analyze-changes.outputs.changes-externals}}
       changes-c-cpp: ${{needs.analyze-changes.outputs.changes-c-cpp}}
       changes-md: ${{needs.analyze-changes.outputs.changes-md}}
+      use-sccache: ${{github.actor != 'dependabot[bot]'}}
+      sccache-write: false
```
Comment (devpow112) [inline .github/workflows/pull-request.yml:140]: Only allowing read during pull request runs. This seems to be causing builds to stall/take way longer if there are multiple concurrent PR's running.

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -14,3 +14,5 @@ jobs:
       run-check-licenses: true
       run-format-check-c-cpp: true
       run-format-check-md: true
+      use-sccache: ${{github.actor != 'dependabot[bot]'}}
+      sccache-write: true
```
Comment (devpow112) [inline .github/workflows/push.yml:18]: Once stuff merges then we write the new objects.

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -14,3 +14,5 @@ jobs:
       run-check-licenses: true
       run-format-check-c-cpp: true
       run-format-check-md: true
+      use-sccache: ${{github.actor != 'dependabot[bot]'}}
```
Comment (devpow112) [inline .github/workflows/push.yml:17]: Secrets aren't available during dependabot runs anyways so sccache doesn't actually get used correctly

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -136,3 +136,5 @@ jobs:
       changes-externals: ${{needs.analyze-changes.outputs.changes-externals}}
       changes-c-cpp: ${{needs.analyze-changes.outputs.changes-c-cpp}}
       changes-md: ${{needs.analyze-changes.outputs.changes-md}}
+      use-sccache: ${{github.actor != 'dependabot[bot]'}}
```
Comment (devpow112) [inline .github/workflows/pull-request.yml:139]: Secrets aren't available during dependabot runs anyways so sccache doesn't actually get used correctly

File: .github/workflows/push.yml (hunk 2 above)
Comment (AWoloszyn) [reply .github/workflows/push.yml:18]: I think that makes a lot of sense anyway. It's unlikely that prs will depend on other prs so we won't get a speedup anyway

**Review by AWoloszyn — DISMISSED** [review]

**Review by AWoloszyn — APPROVED** [review]

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -128,6 +128,7 @@ jobs:
     needs:
       - analyze-changes
     uses: ./.github/workflows/ci.yml
+    secrets: inherit
```
Comment (devpow112) [inline .github/workflows/pull-request.yml:131]: Secrets don't get passed by default. This was why our build times have balloned.

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -9,8 +9,11 @@ jobs:
   ci:
     name: CI
     uses: ./.github/workflows/ci.yml
+    secrets: inherit
```
Comment (devpow112) [inline .github/workflows/push.yml:12]: Secrets don't get passed by default. This was why our build times have balloned.

**Review by AWoloszyn — APPROVED** [review]

**Pull request merged**